skfair/metrics/fairness_report.py

```diff
@@ -0,0 +1,117 @@
+from functools import partial
+from collections import defaultdict
+
+from sklearn.metrics import (confusion_matrix, f1_score, precision_score,
+                             accuracy_score, recall_score)
+import numpy as np
+import pandas as pd
+
+
+def false_positive_score(y_true, y_pred, labels=None):
+    conf_matrix = confusion_matrix(y_true, y_pred, labels=labels)
+    tn, fp, fn, tp = true_false_positive_negative(conf_matrix)
+    eps = 1e-10
+    return fp / (fp + tn + eps)
+
+
+def false_discovery_score(y_true, y_pred, labels=None):
+    conf_matrix = confusion_matrix(y_true, y_pred, labels=labels)
+    tn, fp, fn, tp = true_false_positive_negative(conf_matrix)
+    eps = 1e-10
+    return fp / (tp + fp + eps)
+
+
+DEFAULT_METRICS = {
+    "TPR": partial(recall_score, average="micro"),
+    "FPR": false_positive_score,
+    "PPVR": partial(precision_score, average="micro"),
+    "FDR": false_discovery_score,
+    "ACC": accuracy_score,
+    "F1": partial(f1_score, average="micro")
+}
+
+
+def true_false_positive_negative(conf_matrix):
+    # conf matrix: rows are true, cols are pred
+    if conf_matrix.shape == (2, 2):  # binary case
+        tn, fp, fn, tp = conf_matrix.ravel()
+        return tn, fp, fn, tp
+    tp = 0
+    fp = 0
+    fn = 0
+    tn = 0
+    # A second check on the below is appreciated.
+    for i in range(conf_matrix.shape[0]):
+        tp += conf_matrix[i, i]
+        fn += conf_matrix[i, :].sum() - conf_matrix[i, i]
+        fp += conf_matrix[:, i].sum() - conf_matrix[i, i]
+        tn += (conf_matrix[:i-1, :i-1].sum() + conf_matrix[:i-1, i+1:].sum() +
+               conf_matrix[i+1:, :i-1].sum() + conf_matrix[i+1:i+1:].sum())
+    return tn, fp, fn, tp
+
+
+def yield_metrics(metrics):
+    if type(metrics) == list:
+        for metric in metrics:
+            yield metric.__name__, metric
+    else:
```

Comment:
Might be nicer to check for `list` and `dict` and to throw an error otherwise. This is also a function that can be hidden away from the end-user.

Reply:
You mean by adding an underscore in the start of the name?